src/llamafactory/data/mm_plugin.py

```diff
@@ -417,11 +417,55 @@ def get_mm_inputs(
         return self._get_mm_inputs(images, videos, processor)
 
 
+class Llama3VlPlugin(BasePlugin):
+    @override
+    def process_messages(
+        self,
+        messages: Sequence[Dict[str, str]],
+        images: Sequence["ImageInput"],
+        videos: Sequence["VideoInput"],
+        processor: Optional["ProcessorMixin"],
+    ) -> List[Dict[str, str]]:
+        self._validate_input(images, videos)
+        num_image_tokens = 0
+        messages = deepcopy(messages)
+        for message in messages:
+            content = message["content"]
+            while IMAGE_PLACEHOLDER in content:
+                num_image_tokens += 1
+                content = content.replace(IMAGE_PLACEHOLDER, "<|image|>", 1)
+
+            message["content"] = content
+
+        if len(images) != num_image_tokens:
+            raise ValueError("The number of images does not match the number of {} tokens".format(IMAGE_PLACEHOLDER))
+
+        return messages
+
+    def get_mm_inputs(
+        self,
+        images: Sequence["ImageInput"],
+        videos: Sequence["VideoInput"],
+        imglens: Sequence[int],
+        vidlens: Sequence[int],
+        seqlens: Sequence[int],
+        processor: Optional["ProcessorMixin"],
+    ) -> Dict[str, Union[List[int], "torch.Tensor"]]:
+        super().get_mm_inputs(images, videos, imglens, vidlens, seqlens, processor)
+        if images is not None:
+            images = [Image.open(image) if isinstance(image, str) else image for image in images]
+            image_features = processor.image_processor(images)
+            _ = image_features.pop("num_tiles")
+        image_features = {k: v if isinstance(v, torch.Tensor) else torch.tensor(v) for k, v in image_features.items()}
```

Comment:
we need to add cross attention masks https://github.com/huggingface/transformers/blob/0a21381ba3047882ffe1b95c639aec28974b2c7e/src/transformers/models/mllama/processing_mllama.py#L322-L333

Reply:
That is because we can't get text at `get_mm_inputs` how do you think to fix this?

Reply:
yep, we should do some work here